core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1851,4 +1851,27 @@ private void unparseHashrowFunction(SqlWriter writer, SqlCall call, int leftPrec
         farmFingerprintOperandCall);
     super.unparseCall(writer, farmFingerprintCall, leftPrec, rightPrec);
   }
+
+  private SqlWriter.Frame getTruncFrame(SqlWriter writer, SqlCall call) {
+    SqlWriter.Frame frame = null;
+    String dateFormatOperand = call.operand(1).toString();
+    boolean isDateTimeOperand = call.operand(0).toString().contains("DATETIME");
```

Comment:
What about timestamp with time zone , as we cast only in case of timetsamp without time zone.

Reply:
As discussed , we are still converting 'timestamp with time zone' to 'datetime' , Ideally we  should convert it to Timestamp only.
So we can take care of this bigquery change in a separate jira.